src/Features/Core/AddMissingReference/AbstractAddMissingReferenceCodeFixProvider.cs

```diff
@@ -0,0 +1,118 @@
+    {
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var root = await context.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var model = await context.Document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var uniqueIdentities = new HashSet<AssemblyIdentity>();
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                var nodes = FindNodes(root, diagnostic);
+                var types = GetTypesForNodes(model, nodes, cancellationToken).Distinct();
+                AssemblyIdentity identity = GetAssemblyIdentity(types);
+                if (identity != null && !uniqueIdentities.Contains(identity) && !identity.Equals(model.Compilation.Assembly.Identity))
+                {
+                    uniqueIdentities.Add(identity);
+                    context.RegisterCodeFix(
+                        await AddMissingReferenceCodeAction.CreateAsync(context.Document.Project, identity, context.CancellationToken).ConfigureAwait(false),
+                        diagnostic);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Find all the identifier names in the given location, any of these could be the symbols triggering the diagnostic.
+        /// </summary>
+        private static IEnumerable<SyntaxNode> FindNodes(SyntaxNode root, Diagnostic diagnostic)
+        {
+            var node = root.FindNode(diagnostic.Location.SourceSpan);
+            return node.DescendantNodesAndSelf().OfType<TIdentifierNameSyntax>();
+        }
+
+        /// <summary>
+        /// Get all of the symbols we can for the given nodes since we have no way to determine up front which symbol is triggering the error.
+        /// </summary>
+        private static IEnumerable<ITypeSymbol> GetTypesForNodes(SemanticModel model, IEnumerable<SyntaxNode> nodes, CancellationToken cancellationToken)
+        {
+            var symbols = new List<ITypeSymbol>();
+            foreach (var node in nodes)
+            {
+                symbols.Add(model.GetTypeInfo(node, cancellationToken).Type);
+                var symbol = model.GetSymbolInfo(node, cancellationToken).GetAnySymbol();
+                symbols.AddRange(GetTypesFromSymbol(symbol));
+                symbols.AddRange(GetTypesFromSymbol(symbol?.OriginalDefinition));
+            }
+
+            return symbols;
+        }
+
+        /// <summary>
+        /// Look for additional symbols related to the symbol that we have.
+        /// All of these are candidates for the IErrorTypeSymbol that is causing the missing assembly reference.
+        /// </summary>
+        private static IEnumerable<ITypeSymbol> GetTypesFromSymbol(ISymbol symbol)
+        {
+            if (symbol != null)
+            {
+                if (symbol is IMethodSymbol)
```

Comment:
There's a ISymbolExtensions.GetParameters that checks if it's a method or property symbol and returns the appropriate stuff.